Documentation/policy-comments.md

```diff
@@ -0,0 +1,169 @@
+ * we can add comments later.
+
+All these reasons and other excuses why the code is not commented make it harder for the future reader of the code 
+to understand it. Yes, it is true that you should always try to write the code as if you could not write any comment to it. 
+Having a well defined [code style](./coding-style.md) helps with that. But regardless how good your code is, you still need 
+to add comments. Writing comments captures intent and this is something that raw code can not do. If the code is buggy 
+and there is no comment anywhere, how can you tell that the code is buggy? If you are not the one who wrote it and you have 
+no comments, you can only guess what was the real intent of the author.
+
+The design of the code and the intentions of its author are important knowledge and the knowledge has to be shared. 
+Otherwise, it is hard for everyone to work with the code if the only person with that knowledge is not available.
+Spend the time now to add comments to save the time to your collegues, or your future self, later. If you do that, you will 
+find out that in order to create a good comment, you have to understand what you are doing. By trying to explain your intention, 
+you will often find a problem in your design or its implementation and you will actually be able to complete your task faster.
+
+If you are interested to learn more about why to comment your code, there are almost unlimited number of resources on this 
+topic on the Internet. You will find many blogs and even whole books talking about the necessity of comments.
+
+
+## Rules
+
+We have two types of rules. Hard rules are those that are required and your code will not be accepted if it violates 
+them. Soft rules are more like guidelines - try to use them where possible. The reviewer can ask you to update your code 
+if it violates a soft rule and it can be fixed easily.
+
+ * **XMLDOC** - Use triple slash (XML documentation) comments for classes, interfaces, methods, parameters, fields, properties, structures, enums, etc. These comments are required - even for trivial code. *(HARD)*
+   * **SUMMARY-1** - First paragraph in `<summary>` must be short, no more than 3 lines. If you find you need more than 2 lines, maybe you are not writing the summary anymore and you are already explaining details. *(HARD)*
+   * **SUMMARY-2** - `<summary>` should contain information useful for someone who is trying to understand the code. Implementation details belong to `<remarks>`. *(SOFT)*
+   * **METHODS** - Method comments must contain `<summary>`. If the method has input and/or output parameters `<param>` must be used for each of them. If the method returns anything else than `void` or `Task` (but not `Task<T>`),
+it must contain `<returns>`. The order of the tags is `<summary>`, `<param>`, `<returns>`, `<exception>`, `<remarks>`, `<example>`, `<seealso>`. *(HARD)*
```

Comment:
`<param>` should be always documented. Yes, some times this is prone to triviality, but it again helps you thinking about what the param is for, are there any catches. Also helps with IntelliSense. So I would be strict with `<param>`.